app/controllers/Application.java

```diff
@@ -430,6 +430,7 @@ public static Result flowHistory() {
     if (results.size() == 0) {
       return notFound("Unable to find record on flow url: " + flowDefId);
     }
+    IdUrlPair flowDefPair = new IdUrlPair(flowDefId, results.get(0).flowDefUrl);
```

Comment:
Pretty much everywhere that the UI displays a `flowDefId` the text is a link to the relevant page in the scheduler so we need the URL to create the link. Technically yes, there is sufficient information to derive the link from only the `flowDefId`, however doing so would require per-scheduler logic in either the controller or view code (and as far as I can tell the whole reason for storing URLs in the database is to avoid this situation exactly). Using a pair seems to me to be the cleanest way of doing things.